### PR TITLE
Add GrabPay to PaymentSheet

### PR DESCRIPTION
**Pull request by porter-stripe**
## Summary
- Adds GrabPay for PI's to PaymentSheet

https://github.com/stripe/stripe-ios/assets/88012362/6d6d8938-57c9-46f2-b09c-e4dfdc368c04

## Motivation
https://github.com/stripe/stripe-ios/issues/2672

## Testing
- Manual in test mode
- Additional unit test

## Changelog
See diff

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentMethodType.swift [hunk 1]
```diff
@@ -42,6 +42,7 @@ extension PaymentSheet {
         case dynamic(String)
         case UPI
         case cashApp
+        case grabPay
```
Comment (porter-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentMethodType.swift:45]: Adding LPM's would be a bit easier if we ripped this type out, as we now have two types representing the ~same thing.

**Comment by porter-stripe** [comment]
Blocked on https://git.corp.stripe.com/stripe-internal/sandbox-apps/pull/523

Need to deploy once ^ is merged

File: Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetUITest.swift [hunk 2]
```diff
@@ -663,6 +662,37 @@ class PaymentSheetStandardLPMUITests: PaymentSheetUITestCase {
         // no pay button tap because linked account is stubbed/fake in UI test
     }
 
+    func testGrabPayPaymentMethod() throws {
```
Comment (yuki-stripe) [inline Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetUITest.swift:665]: Let's add a test to `PaymentSheetLPMConfirmFlowTests.swift`, should be super simple and we'll get full confirm flow coverage.

**Review by yuki-stripe — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## x.x.x x-x-x
+### PaymentSheet
+* [Added] Support for GrabPay with PaymentIntent's.
```
[the remark this answers is not in the document]
Comment (porter-stripe) [reply CHANGELOG.md:3]: https://github.com/stripe/stripe-ios/pull/2851

**Pull request merged**